### PR TITLE
Update the DNS spec from upstream

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:
Updates kubeadm to use the latest DNS spec.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

@bowei @MrHohn @thockin  In the future, kubedns changes should be applied to this kubeadm file as well

**Release note**:

```release-note
NONE
```

@pires @errordeveloper @dmmcquay @mikedanese

### CODE REVIEW

File: cmd/kubeadm/app/phases/addons/manifests.go [hunk 1]
```diff
@@ -71,9 +71,8 @@ spec:
       - name: kube-proxy
         image: {{ .Image }}
         imagePullPolicy: IfNotPresent
-        # TODO: This is gonna work with hyperkube v1.6.0-alpha.2+: https://github.com/kubernetes/kubernetes/pull/41017
         command:
-        - kube-proxy
+        - /usr/local/bin/kube-proxy
```
Comment (pires) [inline cmd/kubeadm/app/phases/addons/manifests.go]: Why is this related to DNS? 🙄

**Review by pires — CHANGES_REQUESTED** [review]
Other than that, lgtm.

File: cmd/kubeadm/app/phases/addons/manifests.go (hunk 1 above)
Comment (luxas) [inline cmd/kubeadm/app/phases/addons/manifests.go]: Ok, this came at the same time when cherry-picking, but it's okay.
Basically, there is a chance that we won't have shells in the future, so we should point at the absolute path (which now is the same for kube-* images and hyperkube thanks to a PR I made last week)

cc @timstclair and ref: https://github.com/kubernetes/kubernetes/issues/40248

**Review by pires — APPROVED** [review]
